### PR TITLE
Skills - Rally

### PR DESCRIPTION
**Pull request by JesterWizard**
Rally skills: Boost the respective stat of any allies in a 2x2 range of the skill holder by +4

I made each of the rally skills a seperate file, not sure if that's best practice or not?

### CODE REVIEW

**Comment by MokhaLeee** [comment]
too many conflicts

**Comment by MokhaLeee** [comment]
And these menu effects are lacking in target-selection. You should spend time on studying vanilla target-selection routine, which should also modify on usablity routine, they also should use a MakeTargetListxxx API as what vanilla did.

**Comment by MokhaLeee** [comment]
https://fireemblem.fandom.com/wiki/Rally

I think the rally skill should be consistent with the original version, allowing the character to choose a target to give buff. If this is difficult for you to implement, I think you can submit the skillinfo first.

**Comment by JesterWizard** [comment]
The original implementation **_is_** as I've PR'd it, as per the wiki link you posted. You're thinking of the Three Houses implementation.

File: Wizardry/Core/SkillSys/MiscSkillEffects/MenuSkills/MiscSkillEffects.event [hunk 1]
```diff
@@ -0,0 +1,51 @@
+#include "PostActionSkills/AlertStance.lyn.event"
```
Comment (MokhaLeee) [inline Wizardry/Core/SkillSys/MiscSkillEffects/MenuSkills/MiscSkillEffects.event:1]: I think this file should not be modified like this?

File: Wizardry/Core/SkillSys/MiscSkillEffects/MenuSkills/RallyDefense.c [hunk 2]
```diff
@@ -0,0 +1,77 @@
...
+u8 RallyDefense_Usability(const struct MenuItemDef * def, int number)
+{
+    if (gActiveUnit->state & US_CANTOING)
+        return MENU_NOTSHOWN;
+
+    int i;
+    bool allies_in_range = false;
+
+    for (i = 0; i < ARRAY_COUNT_RANGE2x2; i++)
+            {
```
Comment (MokhaLeee) [inline Wizardry/Core/SkillSys/MiscSkillEffects/MenuSkills/RallyDefense.c:18]: code style

**Comment by MokhaLeee** [comment]
Okay, I have some idea to optimize them, this PR may be merged after the commits has been solved

File: Wizardry/Core/SkillSys/MiscSkillEffects/MenuSkills/MiscSkillEffects.event (hunk 1 above)
Comment (JesterWizard) [reply Wizardry/Core/SkillSys/MiscSkillEffects/MenuSkills/MiscSkillEffects.event:1]: Well the lyn files are needed here. I did state originally that I wasn't sure if seperate C files were needed for each rally skill or not. I wasn't sure of the best practice in this scenario.

File: Wizardry/Core/SkillSys/MiscSkillEffects/MenuSkills/RallyDefense.c (hunk 2 above)
Comment (JesterWizard) [reply Wizardry/Core/SkillSys/MiscSkillEffects/MenuSkills/RallyDefense.c]: Fixed

File: Wizardry/Core/SkillSys/MiscSkillEffects/MenuSkills/MiscSkillEffects.event (hunk 1 above)
Comment (MokhaLeee) [reply Wizardry/Core/SkillSys/MiscSkillEffects/MenuSkills/MiscSkillEffects.event:1]: image
so what is this

Comment (MokhaLeee) [reply Wizardry/Core/SkillSys/MiscSkillEffects/MenuSkills/MiscSkillEffects.event:1]: Fine I have identified this

Comment (JesterWizard) [reply Wizardry/Core/SkillSys/MiscSkillEffects/MenuSkills/MiscSkillEffects.event:1]: Sorry Mokha, I'm not sure what you're asking me, have I done something wrong?

My understanding has been the lyn files need to be included here to generate them for the rally skills.

**Pull request merged**